modules/@ergonode/products/src/components/Tabs/ProductCatalogTab.vue

```diff
@@ -55,17 +59,25 @@ export default {
 
             this.setVerticalTabs();
         },
+        layout() {
+            this.setVerticalTabs();
+        },
     },
     mounted() {
         this.setVerticalTabs();
     },
     methods: {
+        onLayoutChange(layout) {
+            this.layout = layout;
+        },
         async setVerticalTabs() {
             const extendedVerticalTabs = await this.$getExtendMethod('@Products/components/Tabs/ProductCatalogTab/verticalTabs', {
                 $this: this,
             });
 
-            this.verticalTabs = [].concat(...extendedVerticalTabs);
+            this.verticalTabs = []
+                .concat(...extendedVerticalTabs)
+                .filter(tab => typeof tab.layouts === 'undefined' || tab.layouts.some(layout => layout === this.layout));
```

Comment:
Maybe it would be worth applying -> js optional chaining
.filter(tab => tab.layouts?.some(layout => layout === this.layout));